### ISSUE TITLE
IDE0059 erroneously raised for write into span

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#77390]
**Version Used**: 
Version 17.14.0 Preview 2.0 [35820.302.main]

**Steps to Reproduce**:
```C#
internal class Program
{
    private static void Main()
    {
        byte[] buffer = new byte[1];
        Write(buffer, 0);
        Console.WriteLine(buffer[0]);
    }

    private static void Write(Span<byte> buffer, int i)
    {
        buffer[i] = 0x7F;
    }
}
```

**Diagnostic Id**:
IDE0059

**Expected Behavior**:
No diagnostics.

**Actual Behavior**:
IDE0059 is triggered on the `buffer[i] = 0x7F` line, even though it's necessary.
![Image](https://github.com/user-attachments/assets/5840ac7f-2518-46de-91fa-a9835cfcc5eb)

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/77297
